src/Timer.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { TimerProps, TimerState } from './models/models';
+
+const START_TIME = 100;
+
+class Timer extends React.Component<TimerProps, TimerState> {
+
+  state = { time: { minutes: 1, seconds: 0 }, songCount: 1};
+
+  constructor(props: TimerProps) {
+    super(props);
+    this.startTimer = this.startTimer.bind(this);
+  }
+
+  componentDidMount() {
+    this.startTimer();
+  }
+
+  async startTimer() {
+    // Loop through a list of [0,1,2,3...START_TIME-1]
+    for (const num of Array(START_TIME).keys()) {
+      // Set a tick to occur every 1.000 * n seconds
+      await setTimeout(this.tick, 1000*num, num);
+    }
+  }
+
+  tick = async (num: number) => {
+    // Unroll this.state.time into new object (basically clone it)
+    let newTime = { ...this.state.time };
+    if(this.state.time.seconds === 0) {
+      if(this.state.time.minutes === 0) {
+        this.props.skipToNextSong();
+        return;
+      }
+      newTime.minutes -= 1;
```

Comment:
If we're here then we know `seconds = 0` but `minutes != 0`, so reduce a minute and add 60 s

Reply:
Not sure if you saw my updates to the README in the `main` brach, but I'd like to implement a pause/play functionality to the Timer component. What are your thoughts on that?

Reply:
For sho, I like that. Let me finish up this major refactor then we can raise a few `Issues` against the Repo that specify features we want, and each of us can raise PRs to fix the "issues"